### PR TITLE
vNext Build process cleanup for newer tooling

### PR DESCRIPTION
**Pull request by attilah**
Reduce logic in build.cmd
Enable auto generation of AssemblyInfo.cs
Remove Version.txt
Remove GlobalAssemblyInfo.cs
Bump DotnetCLIVersion.txt
Cleanup AssemblyInfo.cs files

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,8 +2,9 @@
 ## files generated by popular Visual Studio add-ons.
 
 # Tool Runtime Dir
-/vNext/src/[Tt]ools/
+/vNext/[Tt]ools/
 /vNext/src/[Bb]ootstrap/
```
Comment (ReubenBond) [inline .gitignore]: is it using src/Bootstrap as well as /Bootstrap? fine either way

File: vNext/Build.cmd [hunk 2]
```diff
@@ -52,9 +38,13 @@ call %_dotnet% build %BUILD_FLAGS% /p:ArtifactDirectory=%OutputPath%\;Configurat
 @if ERRORLEVEL 1 GOTO :ErrorStop
 @echo BUILD ok for %CONFIGURATION% %SOLUTION%
 
-call %_dotnet% pack %PACK_FLAGS% /p:ArtifactDirectory=%OutputPath%\;Configuration=%CONFIGURATION% "%SOLUTION%"
+call %_dotnet% restore "%CodeGenProject%"
+@if ERRORLEVEL 1 GOTO :ErrorStop
+@echo BUILD ok for %CONFIGURATION% %CodeGenProject%
```
Comment (ReubenBond) [inline vNext/Build.cmd]: Seems this should say "RESTORE ok for..."

Comment (attilah) [inline vNext/Build.cmd]: Noone reads these :-)

I don't see this as an issue, but if you insist I can update it.

File: .gitignore (hunk 1 above)
Comment (attilah) [inline .gitignore]: I am not sure. We will need to clean this part up when merging the 2 trees I think.

File: vNext/Orleans.vNext.sln [hunk 3]
```diff
@@ -422,184 +405,172 @@ Global
...
 		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x64.ActiveCfg = Debug|x64
-		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x64.Build.0 = Debug|x64
-		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x86.ActiveCfg = Debug|x86
-		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x86.Build.0 = Debug|x86
+		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x64.Build.0 = Debug|Any CPU
+		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{D9718C48-7020-4A0F-ACA0-17B28DCD6339}.Debug|x86.Build.0 = Debug|Any CPU
```
Comment (jdom) [inline vNext/Orleans.vNext.sln]: There seem to be a lot of unnecessary configurations. In a future PR we should probably clean those

File: vNext/src/OrleansAWSUtils/OrleansAWSUtils.csproj [hunk 4]
```diff
@@ -3,9 +3,7 @@
     <IsPackable>true</IsPackable>
     <PackageId>Microsoft.Orleans.OrleansAWSUtils</PackageId>
     <Title>Microsoft Orleans AWS Utilities</Title>
-    <Description>
-      Library of utility types for Amazon AWS of Microsoft Orleans.
-    </Description>
+    <Description>Library of utility types for Amazon AWS of Microsoft Orleans.</Description>
```
Comment (ReubenBond) [inline vNext/src/OrleansAWSUtils/OrleansAWSUtils.csproj]: Not your fault, but if you're okay making revisions to this, could we write:
`Support library for hosting Orleans on Amazon AWS`

That way it matches the description for Azure utils

File: vNext/Directory.Build.props [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+  </PropertyGroup>
 
-    <!-- Temporary properties until NuGet bug will be fixed: https://github.com/NuGet/Home/issues/4508 -->
-    <!-- Keep in sync with Build\Version.txt -->
-    <VersionPrefix>2.0.0.0</VersionPrefix>
-    <VersionSuffix>preview1</VersionSuffix>
+  <!-- For Debug builds generated a date/time dependent version suffix -->
+  <PropertyGroup Condition=" '$(Configuration)'=='Debug' ">
+    <VersionSuffix Condition=" '$(VersionSuffix)'=='' ">dev</VersionSuffix>
+    <VersionSuffix>$(VersionSuffix)-$([System.DateTime]::Now.ToString(yyyyMMddmmss))</VersionSuffix>
```
Comment (jdom) [inline vNext/Directory.Build.props]: I believe these are being lazily evaluated per project, so it generates projects that do no align with each other. We should find a way to make it global for the entire build. I'll merge for now and you can submit a separate PR for it.

**Comment by ReubenBond** [comment]
Woo :D Great work, @attilah 🙌

File: vNext/Directory.Build.props (hunk 5 above)
Comment (attilah) [inline vNext/Directory.Build.props]: You mean that it is evaluated on a per project basis, so datetime value can be different for 1 build?

I will look into it, can be a nice catch.

Comment (jdom) [inline vNext/Directory.Build.props]: Indeed

Comment (attilah) [inline vNext/Directory.Build.props]: Fixed ;-)

**Pull request merged**